python/packages/isce3/signal/interpolate_by_range.py

```diff
@@ -1,44 +1,80 @@
 import numpy as np
 
+
 def decimate_freq_a_array(
         slant_main,
         slant_side,
         target_runw):
-    """decimate target_runw of main band to have same size with side band
-    assuming slant_main and slant_side are evenly spaced
+    """Decimate target_runw of main band to have same size as side band,
+    assuming slant_main and slant_side are evenly spaced.
 
     Parameters
     ----------
     slant_main : numpy.ndarray
-        slant range array of frequency A band
+        Slant range array of frequency A band
     slant_side : numpy.ndarray
-        slant range array of frequency B band
+        Slant range array of frequency B band
     target_runw : numpy.ndarray
-        RUNW array of frequency A band
-        width of target_runw should be same with length of slant_main
+        RUNW array of frequency A band.
+        Width of target_runw should be same as length of slant_main.
 
     Returns
     -------
     decimated_array : numpy.ndarray
-        decimated RUNW array
+        Decimated RUNW array with width == len(slant_side).
     """
     _, width = target_runw.shape
 
     first_index = np.argmin(np.abs(slant_main - slant_side[0]))
     spacing_main = slant_main[1] - slant_main[0]
     spacing_side = slant_side[1] - slant_side[0]
 
-    resampling_scale_factor = int(np.round(spacing_side / spacing_main))
+    # make sure stride is at least 1
+    resampling_scale_factor = max(1,
+                                  int(np.round(spacing_side / spacing_main)))
```

Comment:
```suggestion
    resampling_scale_factor = max(
        1, int(np.round(spacing_side / spacing_main))
    )
```
Just a little stylistic recommendation, I'm not a big of these heavily indented argument lists because they condense all the parameters into a little chunk way over to the right.

Reply:
Thanks. @Tyler-g-hudson